prometheus-metric-storage/src/lib.rs

```diff
@@ -50,6 +53,14 @@ impl StorageRegistry {
     /// Convert this wrapper into the underlying [`Registry`].
     ///
     /// All information about registered storages is lost.
+    // nlordell: Doesn't this potentially drop the `storages` map and therefore
+    // the `Pin<Box<...>>`s static reference no longer valid? For example:
+    // ```
+    // let storage = StorageRegistry::new(...);
+    // let x: &'static MyMetrics = storage.get_or_create_storage(...);
+    // let _ = storage.into_registry();
+    // *x; // UB - use after free
```

Comment:
This wouldn't compile because we're dropping storage while it's borrowed.

Reply:
Because the reference it gets out is tied to the storage and not 'static, right?

Reply:
Yes, exactly

Reply:
And if the storage itself is `'static`, then it's
`unsafe` to drop it.

Reply:
Yeah makes sense. I had just remembered that we had `'static` references in services and didn't even check that the returned reference is tied to `&self` :see_no_evil:.